cohort-engine/src/main/java/com/ibm/cohort/engine/CqlEngineWrapper.java

```diff
@@ -164,6 +164,15 @@ public void setTerminologyServerClient(IGenericClient client) {
 	public IGenericClient getTerminologyServerClient() {
 		return this.terminologyServerClient;
 	}
+	
+	/**
+	 * Memory leaks can occur when holding threads are re-used which is the case on application servers using pool of threads.
+	 * 
+	 * This method would allow the ThreadLocal variable to be cleaned up in such a scenario.
+	 */
+	public void unloadModelResolver() {
```

Comment:
> Call "remove()" on "MODEL_RESOLVER".

> ThreadLocal variables are supposed to be garbage collected once the holding thread is no longer alive. Memory leaks can occur when holding threads are re-used which is the case on application servers using pool of threads.  To avoid such problems, it is recommended to always clean up ThreadLocal variables using the remove() method to remove the current thread’s value for the ThreadLocal variable.